### PR TITLE
Feature/upmcfhir 241 add category to comp

### PR DESCRIPTION
**Pull request by 5t33**
#### What does this PR do?
Adds:
- composition.class
- composition.section.note

#### Related JIRA tickets:
[UPMCFHIR-262](https://jira.amida.com/browse/UPMCFHIR-262)
[UPMCFHIR-241](https://jira.amida.com/browse/UPMCFHIR-241)

### CODE REVIEW

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 1]
```diff
@@ -768,6 +778,11 @@ public EntryResult tClinicalDocument2Bundle(ClinicalDocument cdaClinicalDocument
 			fhirComp.setId(new IdType("Composition", getUniqueId()));
 			result.addResource(fhirComp);
 
+			CodeableConcept classConcept = new CodeableConcept();
```
Comment (5t33) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:781]: Here's where I added the class to the comp

File: src/test/java/tr/com/srdc/cda2fhir/CompositionTest.java [hunk 2]
```diff
@@ -15,50 +15,57 @@
...
 		Assert.assertEquals("Expect title to equal title", defaultExpectedTitle, comp.getTitle());
-		Assert.assertEquals("Expect confidentiality to eqial confidentiality", defaultExpectedConfidentiality, comp.getConfidentiality().toString());
-
+		Assert.assertEquals("Expect confidentiality to equal confidentiality", defaultExpectedConfidentiality,
+				comp.getConfidentiality().toString());
+		Assert.assertEquals("Expect composition to code system to equal http://hl7.org/fhir/ValueSet/doc-classcodes",
+				"http://hl7.org/fhir/ValueSet/doc-classcodes", comp.getClass_().getCodingFirstRep().getSystem());
+		Assert.assertEquals("Expect composition to code to equal LP173418-7", "LP173421-7",
+				comp.getClass_().getCodingFirstRep().getCode());
+		Assert.assertEquals("Expect composition to code display to equal Note", "Note",
```
Comment (5t33) [inline src/test/java/tr/com/srdc/cda2fhir/CompositionTest.java:68]: testing comp class

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 3]
```diff
@@ -3004,9 +3045,12 @@ public SectionComponent tSection2Section(Section cdaSection) {
 				fhirSec.setText(fhirText);
 		}
 
+		fhirSec.setMode(SectionMode.SNAPSHOT);
```
Comment (5t33) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java:3048]: Here's where I set the mode

**Comment by 5t33** [comment]
I didn't add much but the formatting messed with it. I put comments where I changed things and tested.

File: src/test/java/tr/com/srdc/cda2fhir/SectionTest.java [hunk 4]
```diff
@@ -15,72 +16,72 @@
...
 	}
+
+	@Test
+	public void testSectionMode() {
+		Config.setGenerateNarrative(false);
+		Section cdaSection = factories.base.createSection();
+
+		SectionComponent fhirSection = rt.tSection2Section(cdaSection);
+
+		Assert.assertEquals("Expect fhir section mode to equal", SectionMode.SNAPSHOT.toCode(),
```
Comment (5t33) [inline src/test/java/tr/com/srdc/cda2fhir/SectionTest.java:84]: testing for section.mode

**Review by mmccall — APPROVED** [review]
lgtm

**Pull request merged**